.github/workflows/release-sync.yml

```diff
@@ -0,0 +1,162 @@
+            .data.organization.projectV2.id as $projectId |
+            .data.organization.projectV2.field.id as $fieldId |
+            .data.organization.projectV2.field.options[] |
+            select(.name == $column_name) |
+            {
+              projectId: $projectId,
+              fieldId: $fieldId,
+              optionId: .id
+            }')
+          PROJECT_ID=$(echo "$PROJ_DATA" | jq -r '.projectId')
+          FIELD_ID=$(echo "$PROJ_DATA" | jq -r '.fieldId')
+          OPTION_ID=$(echo "$PROJ_DATA" | jq -r '.optionId')
+
+          # Store values in step output
+          echo "PROJECT_ID=$PROJECT_ID" >> "$GITHUB_OUTPUT"
+          echo "FIELD_ID=$FIELD_ID" >> "$GITHUB_OUTPUT"
+          echo "OPTION_ID=$OPTION_ID" >> "$GITHUB_OUTPUT"
+
+      - name: Fetch PR card Id
+        id: proj_cardId
+        env:
+          PR_URL: ${{ steps.create_pull.outputs.PR_URL }}
+        run: |
+          QUERY="
+            query {
+              organization(login: \"$ORG_NAME\") {
+                projectV2(number: $PROJECT_NUMBER) {
+                  items(first: 10) {
+                    nodes {
+                      id
+                      content {
+                        ... on PullRequest {
+                          id
+                          title
+                          number
+                          repository {
+                            owner {
+                              login
+                            }
+                            name
+                          }
+                          url
+                        }
+                      }
+                    }
+                  }
+                }
+             }
+            }"
+          PROJ_RESPONSE=$(gh api graphql -f query="$QUERY")
+
+          CARD_ID=$(echo "$PROJ_RESPONSE" | jq -r --arg pr_url "$PR_URL" '
+            .data.organization.projectV2.items.nodes[]
+            | select(.content.url == $pr_url)
+            | .id')
+
+          # Store value in step output
+          echo "CARD_ID=$CARD_ID" >> "$GITHUB_OUTPUT"
+
+      - name: Move card to correct column
+        id: mutateCard
+        env:
+          PROJECT_ID: ${{ steps.proj_columns.outputs.PROJECT_ID }}
+          FIELD_ID: ${{ steps.proj_columns.outputs.FIELD_ID }}
+          OPTION_ID: ${{ steps.proj_columns.outputs.OPTION_ID }}
+          CARD_ID: ${{ steps.proj_cardId.outputs.CARD_ID }}
+        run: |
+          QUERY="
+          mutation {
+            updateProjectV2ItemFieldValue(
+              input: {
+                projectId: \"$PROJECT_ID\",
+                itemId: \"$CARD_ID\",
+                fieldId: \"$FIELD_ID\",
+                value: {
+                  singleSelectOptionId: \"$OPTION_ID\"
+                }
+              }
+            ) {
+              projectV2Item {
+                id
+              }
+            }
+          }
+          "
+          PROJ_RESPONSE=$(gh api graphql -f query="$QUERY")
```

Comment:
Is it possible that this command will fail? Or return a response indicating failure?

Should we treat this and make the workflow fail to warn us?

Reply:
Good point. I'll add an error handler step

Reply:
After some experimentation, I believe it will better to first release this workflow with the sync functionality and then add the error handling to it.

The main reason is because the workflow already has all the error handling and logging features we need, we just need a better channel to be informed about it: that would be adding an integration with Slack.

Since this PR already has a number of repository secrets, access tokens and configurations to be done, I prefer leaving the additional effort to another PR that will complement this one.

What do you think about this?

Reply:
LGTM